test/ws_deque/stm_ws_deque.ml

```diff
@@ -60,29 +60,30 @@ module WSDConf = struct
     | _, _ -> false
 end
 
-module WSDT_seq = STM_sequential.Make (WSDConf)
-module WSDT_dom = STM_domain.Make (WSDConf)
-
-(* A parallel agreement test - w/repeat and retries combined *)
-let agree_test_par_asym ~count ~name =
-  let rep_count = 20 in
-  let seq_len, par_len = (20, 12) in
-  Test.make ~retries:10 ~count ~name
-    (* "Owner domain" cmds can't be [Steal], "stealer domain" cmds can only be [Steal]. *)
-    (WSDT_dom.arb_triple_asym seq_len par_len WSDConf.arb_cmd WSDConf.arb_cmd
-       WSDConf.stealer_cmd) (fun triple ->
-      assume (WSDT_dom.all_interleavings_ok triple);
-      repeat rep_count WSDT_dom.agree_prop_par_asym triple)
-
 let () =
-  let count = 1000 in
-  QCheck_base_runner.run_tests_main
+  let make_domain ~count ~name
+      (module Dom : Stm_run.STM_domain
+        with type Spec.cmd = Spec.cmd
+         and type Spec.state = Spec.state
+         and type Spec.sut = Spec.sut) =
```

Comment:
This is a case where first-class modules are a bit cumbersome in OCaml, because the syntax for constraints is restricted (i.e. can't specify `(module Dom : STM_run.STM_domain with module Spec = Spec)`), but, on the whole, first-class module perhaps makes things a bit more concise.